### PR TITLE
ci: upgrade typos

### LINKED ISSUE #12165: Reenable typos check
Our CI is broken because our `checkers` Dockerfile cannot compile `typos-cli` v1.3.9. To unblock other work, we have disabled `typos` while upgrading it.

This issue reminds us to reenable `typos` as part of `checkers.sh`.

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #12165

Upgrade `typos-cli`.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12166)

### CODE REVIEW

**Review by alevenberg — APPROVED** [review]

File: ci/cloudbuild/dockerfiles/checkers.Dockerfile [hunk 1]
```diff
@@ -35,7 +35,7 @@ RUN dnf makecache && \
         python-pip \
         ShellCheck
 
-RUN cargo install typos-cli --version 1.3.9 --root /usr/local
+RUN cargo install typos-cli --version 1.16.1 --root /usr/local
```
Comment (devbww) [inline ci/cloudbuild/dockerfiles/checkers.Dockerfile:38]: FWIW, the old version also appears in `doc/contributor/howto-guide-setup-development-workstation.md`.

Comment (dbolduc) [reply ci/cloudbuild/dockerfiles/checkers.Dockerfile:38]: Ack, Thanks. I'll update that doc in a follow up PR, just so this PR can get merged as quickly as possible. :shrug:

Comment (dbolduc) [reply ci/cloudbuild/dockerfiles/checkers.Dockerfile:38]: Fixed.

File: .typos.toml [hunk 2]
```diff
@@ -1,11 +1,15 @@
 [files]
 extend-exclude = [
+    # It does not like "intoto_*.proto"
+    "external/googleapis/protolists/grafeas.list",
```
Comment (coryan) [inline .typos.toml:4]: In a future PR we should we exclude `external/googleapis/protolists/*`?

**Review by coryan — APPROVED** [review]

File: .typos.toml (hunk 2 above)
Comment (dbolduc) [reply .typos.toml:4]: Sure.

**Pull request merged**